subsys/mgmt/osdp/Kconfig

```diff
@@ -71,7 +71,7 @@ config OSDP_SKIP_MARK_BYTE
 
 config OSDP_SC_ENABLED
 	bool "OSDP Secure Channel"
-	depends on CSPRNG_ENABLED
+	depends on CSPRNG_AVAILABLE
```

Comment:
The weirdness here is that we have this set of Kconfigs mixing around the RNG and CSPRNG enablement and needs. 

I was also playing around with a "CSPRNG_AVAILABLE" type of setting but what I think we really need is a "CSPRNG_ENABLE" and "RNG_ENABLE". Because right now we are using the ENABLE_GENERATOR as basically the same configuration which doesn't make sense. If a subsystem/application knows it needs CSRNG or RNG then just set the ENABLE for it and let the Kconfig dependency change get automatically set.

Reply:
> If a subsystem/application knows it needs CSRNG or RNG then just set the ENABLE for it and let the Kconfig dependency change get automatically set.

I don't know if this would solve the problem I addressed with `CSPRNG_AVAILABLE`. As I wrote in the description this work was started by the fact that subsystems maintainers (bluetooth mainly @jhedberg ) were not happy about having to enable `ENTROPY_GENERATOR` (which would be `CSPRNG_ENABLE` if we follow your suggestion) on every board. As far as I understood they just want to say something like "I need a random generator and crypto support and I don't care about how that's implemented in the platform as long as both work properly". This summarizes as "let Mbed TLS enable what it needs in order to get the features that I asked".
The problem is that various subsystems are then tested on a large number of platforms, some of which does not have any capability to generate CSPRNG, but they might still be interested in running the sample/test. For example I just fixed an issue on `zephyr/samples/psa/its/sample.psa.its.secure_storage.entropy_not_secure` which intentionally tests secure storage on platforms that do not have an entropy source. So Mbed TLS "needs to know" what it must enable in order to build properly (is it `ENTROPY_GENERATOR` or `TEST_RANDOM_GENERATOR`?).

Setting `CSPRNG_ENABLE` directly (instead of having it indirectly selected) is another approach and I suspect that this might require us to rework all the samples/tests that then would fail to build. As of now instead this PR does not change any of those (albeit I admit that the work is not 100% done yet, so this might change).

Reply:
I'm not implying doing this on every board. What I'm implying is that every subsystem/application that needs RNG would set directly that it requires CSPRNG or RNG. Let the underlying Kconfig infrastructure handle setting what is needed to accomplish this. 

This is what I don't like about the current system. To get CSPRNG you have to set ENTROPY_GENERATOR... which only makes sense to us doing the entropy or low level RNG stuff.

Reply:
> This is what I don't like about the current system. To get CSPRNG you have to set ENTROPY_GENERATOR

Ok, I see and I also agree that this is not a user friendly. We can definitely improve this.

> What I'm implying is that every subsystem/application that needs RNG would set directly that it requires CSPRNG or RNG. Let the underlying Kconfig infrastructure handle setting what is needed to accomplish this.

I agree also on this, with only a minor note. IMO if we really want to make it generic, we should just add the "enable RNG" Kconfig symbol without specifying explicitly if it's CS or not, then under the hood we do the best we can (i.e. enable CS is there is any way to achieve it, otherwise non-CS). The reason for this is: who would prefer a non-CS source when there is a CS one that claims to work better for the same platform?

Reply:
> I agree also on this, with only a minor note. IMO if we really want to make it generic, we should just add the "enable RNG" Kconfig symbol without specifying explicitly if it's CS or not, then under the hood we do the best we can (i.e. enable CS is there is any way to achieve it, otherwise non-CS). The reason for this is: who would prefer a non-CS source when there is a CS one that claims to work better for the same platform?

I prefer make these options explicit, there are different use cases for rng and csrng. Also if forces user to be thoughtful about the choice.

I would go beyond that, cryptography requires csrng and should enable it, if the target is not capable of that is up the to user to enable a non-cs rng. This would mean in our case that we enable csrng with MBEDTLS (I agree with @jhedberg  subsystem shouldn't need to figure out crypto dependencies by their own, they just want to have crypto ! ), if there is no CSPRNG available the application is responsible to enable a non-csrng by their own.

I don't think we should make exceptions about requirement, if we need CSRNG we need CSRNG. It is up to the user to workaround it because there are security consequences.

Reply:
Ok, looking it more now I think the root cause of the problem is in the random subsys symbols, there is a lot of mixed things between entropy and randomness. Users / applications usually should care about randomness (csrng x rng) and not need to deal with entropy source. Let's discuss / brainstorm it more carefully in the security working group (12/16/2024).